contracts/governance/FeeSharingCollector/FeeSharingCollector.sol

```diff
@@ -782,6 +782,48 @@ contract FeeSharingCollector is
         return amount;
     }
 
+    /**
+     * @dev Get all user fees reward per maxCheckpoint starting from latest processed checkpoint
+     *
+     * @dev e.g: Total user checkpoint for the particualar token = 300,
+     * when we call this function with 50 maxCheckpoint, it will return 6 fee values in array form.
+     * if there is no more fees, it will return empty array.
+     *
+     * @param _user The address of a user (staker) or contract.
+     * @param _token RBTC dummy to fit into existing data structure or SOV. Former address of the pool token.
+     * @param _maxCheckpoints maxCheckpoints to get accumulated fees for the _user
+     * @return The next checkpoint num which is the starting point to fetch all of the fees, array of calculated fees.
+     * */
+    function getAllUserFeesPerMaxCheckpoints(
+        address _user,
+        address _token,
+        uint32 _maxCheckpoints
+    ) external view returns (uint256 nextCheckpointNum, uint256[] memory fees) {
```

Comment:
We decided not to use `getNextPositiveUserCheckpoint` because it will still require the to much iteration on-chainly in order to get the next positive user checkpoint.